federation/backends/minimal_profile.md

```diff
@@ -0,0 +1,175 @@
+# Minimal profile for openEO backends
+This document describes the requirements for a backend that aims to provide minimal functionality for access and processing of gridded
+earth observation data.
+
+This document does not aim to contradict the openEO API, and when this is perceived to be the case, the definitions in the openEO API get precedence.
+It is possible that requirements in this document do define additional constraints on top of the openEO API.
+
+This profile is established because the openEO web service API is quite liberal if interpreted strictly. So a web service can be compliant to the API 
+specification, but still be mostly incompatible with other backends, and existing usage patterns. By following this profile, implementors have a better
+view on what is important.
+
+## Scope
+This minimal profile aims to help initial openEO implementations, that seek a least effort approach to achieve minimal compatibility with other backends.
+Requirements in this profile are not sufficient for backends in the openEO platform federation.
+
+## Mandatory endpoints
+
+In the openEO API, all endpoints are optional and discoverable, except for the basic capabilities and versioning endpoints:
+
+https://api.openeo.org/#tag/Capabilities
+
+https://api.openeo.org/#tag/Capabilities/operation/connect
+
+[Authentication](https://api.openeo.org/#section/Authentication) is also mandatory, but this profile does not define a specific method. We highlight that openID connect + PKCE is one of the only standardized and secure methods that are available.
+ 
+
+
+
+Next to those elements, these are the endpoints that are required by this profile:
+
+
+```
+[
+    {
+      "methods": [
+        "GET"
+      ],
+      "path": "/file_formats"
+    },
+    {
+      "methods": [
+        "GET"
+      ],
+      "path": "/collections"
+    },
+    {
+      "methods": [
+        "GET"
+      ],
+      "path": "/collections/{collection_id}"
+    },
+    {
+      "methods": [
+        "GET"
+      ],
+      "path": "/processes"
+    },
+    {
+      "methods": [
+        "GET",
+        "POST"
+      ],
+      "path": "/jobs"
+    },
+    {
+      "methods": [
+        "GET",
+        "DELETE"
+      ],
+      "path": "/jobs/{job_id}"
+    },
+    {
+      "methods": [
+        "GET",
+        "POST",
+        "DELETE"
+      ],
+      "path": "/jobs/{job_id}/results"
+    }
+  ],
+```
+
+## Batch job support
+
+Support for openEO batch jobs is mandatory, but certain specific elements can be ommitted. The previous section indicates the exact endpoints and HTTP requests that need to be supported.
+
+
+## File formats
+[GeoTiff](./fileformats.md#geotiff) is the only mandatory output format.
+
+## Collections
+A minimal backend shall have at least one raster collection, allowing to build raster data cubes. 
+
+## Processes
+
+### Data Cubes
+
+- `apply`: Apply a process to each pixel
+- `apply_dimension`: Apply a process to pixels along a dimension
```

Comment:
isn't `reduce_dimension` a better candidate for a minimal profile than `apply_dimension`? (especially because there are no UDFs in the minimal profile)

Reply:
Actually, under Math and logic below there are currently no processes that take an array as input (like mean, min, max), so `apply_dimension` nor `reduce_dimension` can not be used anyway

Reply:
I agree, you need reduce_dimension to be able to easily get results for the only supported file format, GeoTiff, in the minimal profile. So reduce_dimension should be included with at least a small number of reducers for composites (min/max/mean?), I'm not sure about apply_dimension.

Reply:
Batch jobs do allow writing to geotiff without reduce_dimension, it just writes a geotiff per date in that case.
But we can indeed switch to reduce_dimension, not sure why I went for apply_dimension...

Reply:
Yes, your implementation can do that and it's allowed, indeed. But that's not minimal for me. Minimal, for me, is providing a 3D cube to write a single GeoTiff.